edge/pkg/devicetwin/dtmanager/membership.go

```diff
@@ -220,7 +218,10 @@ func addDevice(context *dtcontext.DTContext, toAdd []dttype.Device, baseMessage
 		if err != nil {
 			klog.Errorf("Add device %s failed due to some error ,err: %#v", device.ID, err)
 			context.DeviceList.Delete(device.ID)
-			context.Unlock(device.ID)
+			if delta {
+				context.Unlock(device.ID)
+			}
+			context.DeviceMutex.Delete(device.ID)
```

Comment:
If the device is removed from the `DeviceList`, then it should also be removed from the `DeviceMutex`.